### ISSUE TITLE
PT grad accum wrong?

### ISSUE THREAD

**Issue by albertz** [rwth-i6/returnn#1379]
I noticed we have this code:
```python
# scale the loss to account for gradient accumulation
if self._accum_grad_multiple_step > 1:
    for loss_name in train_ctx.losses.keys():
        train_ctx.losses[loss_name].loss /= self._accum_grad_multiple_step
```
This does not make sense to me.

Why do we have this?

This also seems inconsistent to how we have it in TF. We don't do any loss scaling there.

I assume when you have `use_normalized_loss=True` (it's `use_normalized_loss=False` by default), maybe this is the reason to have it? But this is broken in multiple ways. And with `use_normalized_loss=True` (default), it's just wrong?

Also, this looks like it would make the loss accumulation wrong?
```python
losses_dict = NumbersDict(
    {
        name: float(loss.get_summed_loss().raw_tensor.detach().cpu().numpy())
        for name, loss in train_ctx.losses.items()
    }
)
inv_norm_factors_dict = NumbersDict(
    {name: float(_to_raw(loss.get_inv_norm_factor())) for name, loss in train_ctx.losses.items()}
)

accumulated_losses_dict += losses_dict
accumulated_inv_norm_factors_dict += inv_norm_factors_dict
```

This code was added in #1335 (@Judyxujj).

**Comment by albertz** [comment]
So I removed this code now (https://github.com/rwth-i6/returnn/commit/b94b44e5f5bbab9b6f7b39cd1ed3ec556f39635a).

Let's discuss further here in case anything is still wrong now, or further needs to be changed.
